appgate/openapi/parser.py

```diff
@@ -387,30 +418,42 @@ def parse_all_of(self, definitions: List[OpenApiDict]) -> OpenApiDict:
         for d in definitions:
             new_definition["required"].extend(d.get("required", {}))
             new_definition["properties"].update(d.get("properties", {}))
-
-            discriminator = d.get("discriminator")
-            # properties = d.get("properties")
-            if discriminator:
-                new_mapping = {}
-                new_properties = new_definition["properties"]
-                propertyName = discriminator["propertyName"]
-                mapping = discriminator["mapping"]
-                for key, ref in mapping.items():
-                    resolved_ref = self.resolve_reference(ref, [])
-                    if is_compound(resolved_ref):
-                        resolved_ref = self.parse_all_of(resolved_ref["allOf"])
-                    new_mapping.update({key: resolved_ref})
-                    for key, property in resolved_ref["properties"].items():
-                        new_properties.update({key: property})
-                new_definition["discriminator"]["propertyName"] = propertyName
-                new_definition["discriminator"]["mapping"] = new_mapping
-                new_definition["properties"] = new_properties
-
+            new_definition["discriminator"].update(d.get("discriminator", {}))
             if "description" in d:
                 descriptions.append(d["description"])
         new_definition["description"] = ".".join(descriptions)
         return new_definition
 
+    def parse_discriminator(self, definition: Any) -> OpenApiDict:
```

Comment:
A docstring would be nice to explain what kind of transformation this does.
Is definition really `Any`? It looks like its some form of mapping at least.

Reply:
Good catch. Should be `OpenApiDict` type.